kroxylicious-kubernetes-api/src/main/java/io/kroxylicious/kubernetes/api/common/Ref.java

```diff
@@ -0,0 +1,79 @@
+package io.kroxylicious.kubernetes.api.common;
+
+import io.fabric8.kubernetes.api.model.KubernetesResource;
+
+@com.fasterxml.jackson.annotation.JsonInclude(com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL)
+@com.fasterxml.jackson.annotation.JsonPropertyOrder({ "group", "kind", "name" })
+@com.fasterxml.jackson.databind.annotation.JsonDeserialize(using = com.fasterxml.jackson.databind.JsonDeserializer.None.class)
+@javax.annotation.processing.Generated("io.fabric8.java.generator.CRGeneratorRunner")
+@lombok.ToString()
+@io.sundr.builder.annotations.Buildable(editableEnabled = false, validationEnabled = false, generateBuilderPackage = false, builderPackage = "io.fabric8.kubernetes.api.builder", refs = {
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.ObjectMeta.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.ObjectReference.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.LabelSelector.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.Container.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.EnvVar.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.ContainerPort.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.Volume.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.VolumeMount.class)
+})
+public class Ref
```

Comment:
The name Ref makes you think it is a generic reference, but the `listener` really ties it to Strimzi.  References to other Kafka CRs (redpanda or confluent), are likely to need different fields to identify the endpoint that needs to be connected to.

I'm wondering if we'd be better off with:

```yaml
strimziKafkaRef:
   group:
   kind:
   name:
   listenerName:
```

This leaves the door open for:

```yaml
redpandaRef: {}  # a reference to a [redpanda](https://docs.redpanda.com/current/reference/k-crd/#k8s-api-github-com-redpanda-data-redpanda-operator-operator-api-redpanda-v1alpha2-redpanda) 
```   

or a 

```yaml
confluentKafkaRef:  {} # a reference to a https://docs.confluent.io/operator/current/co-api.html#tag/Kafka
```

WDYT?  @SamBarker

Reply:
100% Keith, your model leaves us open to extension which really appeals.